### PR TITLE
Expose more envoy stats by default

### PR DESCRIPTION
**Pull request by bianpengyuan**
context: https://github.com/istio/istio/issues/23166 and https://github.com/istio/istio/issues/23902

This PR creates and exposes more native stats for better observability and debuggability. Specifically stats with the following suffixes will be created:

* Downstream connection stats, `downstream_cx_total`, `downstream_cx_active`, `downstream_cx_http1_total`, `downstream_cx_http2_total`, `downstream_cx_http1_active`, `downstream_cx_http2_active`, `downstream_cx_destroy`, `downstream_cx_destroy_remote`, `downstream_cx_destroy_local`, `downstream_cx_destroy_local_active_rq`, `downstream_cx_destroy_remote_active_rq`

* Upstream connection stats, `upstream_cx_total`, `upstream_cx_active`, `upstream_cx_destroy_local`, `upstream_cx_destroy_remote`, `upstream_cx_destroy_local_with_active_rq`, `upstream_cx_destroy_remote_with_active_rq`, `upstream_cx_close_notify`

* HTTP request route stats, `no_route`, `no_cluster`, `rq_redirect`, `rq_total`, `rq_direct_response`, `rq_reset_after_downstream_response_started`

* Lisener debug stats suffix  `no_filter_chain_match`, `downstream_pre_cx_timeout`

* TCP metadata exchange stats suffix `alpn_protocol_not_found`, `alpn_protocol_found`, `initial_header_not_found`, `header_not_found`, `metadata_added`

An experiment is performed with these stats created by default, and CPU and memory usage diff is negligible: https://snapshot.raintank.io/dashboard/snapshot/21R8BjkvGQCNhevume5pShsJpbZD6Hbx

### CODE REVIEW

**Comment by mandarjog** [comment]
Several customers already enable downstream stats.
Is this using the ip substitution ? Local downstream from client has limited value.

**Comment by bianpengyuan** [comment]
@mandarjog No it does not use ip substitution, but it is cheap to expose local downstream stats with these new suffixes.

File: pkg/bootstrap/instance_test.go [hunk 1]
```diff
@@ -442,12 +457,14 @@ func checkListStringMatcher(t *testing.T, got *matcher.ListStringMatcher, want s
...
 		if pat != "" {
-			patterns = append(patterns, pat)
+			gotPatterns = append(gotPatterns, pat)
 		}
 	}
-	gotPattern := strings.Join(patterns, ",")
-	if want != gotPattern {
-		t.Fatalf("%s mismatch:\ngot: %s\nwant: %s", typ, gotPattern, want)
+
+	sort.Strings(want)
```
Comment (gargnupur) [inline pkg/bootstrap/instance_test.go:464]: why is sort needed? DeepEqual doesn't care about ordering afaik

Comment (howardjohn) [reply pkg/bootstrap/instance_test.go:464]: I think it does for a list

Comment (gargnupur) [reply pkg/bootstrap/instance_test.go:464]: ok.. thanks!

File: pkg/bootstrap/config.go [hunk 2]
```diff
@@ -246,13 +237,40 @@ func getStatsOptions(meta *model.BootstrapNodeMetadata, nodeIPs []string, config
...
+		"downstream_cx_destroy_local", "downstream_cx_destroy_local_active_rq", "downstream_cx_destroy_remote_active_rq",
+		// Upstream connection stats suffix
+		"upstream_cx_total", "upstream_cx_active", "upstream_cx_destroy_local", "upstream_cx_destroy_remote",
+		"upstream_cx_destroy_local_with_active_rq", "upstream_cx_destroy_remote_with_active_rq", "upstream_cx_close_notify",
+		// HTTP request route stats suffix
+		"no_route", "no_cluster", "rq_redirect", "rq_total", "rq_direct_response", "rq_reset_after_downstream_response_started",
+		// Lisener debug stats suffix
+		"no_filter_chain_match", "downstream_pre_cx_timeout",
+		// TCP metadata exchange stats suffix
+		"alpn_protocol_not_found", "alpn_protocol_found", "initial_header_not_found", "header_not_found", "metadata_added",
```
Comment (gargnupur) [inline pkg/bootstrap/config.go:270]: Thanks for adding these!

**Comment by bianpengyuan** [comment]
Put a hold on this since I will need to measure the storage cost of prometheus instance. Might need to trim down this list if it does not look good.

**Pull request closed without merging**